### PR TITLE
Create Resize Observer Meta

### LINKED ISSUE #618: Consider using ResizeObserver
**Enhancement/Discussion**

We might want to consider using a `ResizeObserver` polyfill to allow the `Dimensions` meta manager to perform better, allowing it to detect changes to DOM elements that change the size information of a DOM element.

The draft spec is available at [WICG ResizeObserver Draft](http://rawgit.com/WICG/ResizeObserver/master/index.html).  A well thought out polyfill is available at [que-etc/resize-observer-polyfill](https://github.com/que-etc/resize-observer-polyfill) which uses MutationObservers and Mutation Events to provide spec compliant functionality with some minor limitations.

It is currently available [natively in Chrome/Opera under a flag](https://caniuse.com/#feat=resizeobserver).

Right now, every time a widget is re-rendered, any nodes that are registered with the Dimensions meta manager will have their bounding rectangle re-calculated, whether it has actually been changed.  This would allow the manager to be more reactive and keep a cache of size information, of which its observed DOM elements would have their values updated on actual resize.

Because the polyfill is designed to deliver the resize changes only when it is _convinced_ it has the right information, there could be an issue where the queuing of changes get delivered _after_ the DOM has be re-rendered after an invalidation, either causing a double invalidation or incorrect information being offered up to the widgets.  This would require investigation to help ensure that it operates effectively.

### PR DESCRIPTION
**Pull request by tomdye**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code has been formatted with [`prettier`](https://prettier.io/) as per the [readme code style guidelines](./../#code-style)
* [x] Unit or Functional tests are included in the PR

**Description:**

Creates `ResizeObserver` `meta` which can be used to observe component level resize events.
Requires a global polyfill to be imported when used outside of chrome 64+.

Example deployed [here](https://dist-ajrdioqurf.now.sh/).

Resolves #618

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1176,6 +1176,35 @@ class TestWidget extends WidgetBase<WidgetProperties> {
 }
 ```
 
+#### Resize
+
+The resize observer meta uses the latest [`ResizeObserver`](https://wicg.github.io/ResizeObserver/) within your widgets. This is currently supported within `Chrome 64+` and supported via [polyfill](https://github.com/WICG/ResizeObserver/issues/3).
+
+This allows you to observe resize events at the component level. The `meta` accepts an object of `predicate` functions which receive `ContentRect` dimensions and will be executed when a resize event has occured. The results are made available in your widgets `render` function. This is an incredibly powerful tool for creating responsive layouts.
```
Comment (dylans) [inline README.md]: in your widgets -> in a widget's

File: README.md [hunk 2]
```diff
@@ -1176,6 +1176,35 @@ class TestWidget extends WidgetBase<WidgetProperties> {
 }
 ```
 
+#### Resize
+
+The resize observer meta uses the latest [`ResizeObserver`](https://wicg.github.io/ResizeObserver/) within your widgets. This is currently supported within `Chrome 64+` and supported via [polyfill](https://github.com/WICG/ResizeObserver/issues/3).
```
Comment (dylans) [inline README.md]: within your widgets -> within Dojo-based widgets

Comment (dylans) [reply README.md]: This is currently supported within `Chrome 64+` and supported via -> Native browser support is currently provided by `Chrome 64+` and other Dojo supported browsers work via

**Comment by tomdye** [comment]
@dylans comments addressed and example usage created and deployed [here](https://dist-ajrdioqurf.now.sh/).

**Comment by tomdye** [comment]
@dylans i'll raise an issue on `widgets` relating to the `split-pane` responsiveness

**Comment by tomdye** [comment]
@agubler can I get some 👁 on this please?

**Pull request merged**